sdb/commands/stacks.py

```diff
@@ -216,120 +272,102 @@ def find_module_memory_segment(mod_name: str) -> Tuple[int, int]:
                         mod.core_layout.size.value_())
         return (-1, 0)
 
-    def validate_args(self, args: argparse.Namespace) -> None:
-        if args.function:
-            try:
-                func = sdb.get_object(args.function)
-            except KeyError:
-                raise sdb.CommandError(
-                    self.name,
-                    "symbol '{:s}' does not exist".format(args.function))
-            if func.type_.kind != drgn.TypeKind.FUNCTION:
-                raise sdb.CommandError(
-                    self.name, "'{:s}' is not a function".format(args.function))
-
-        task_states = Stacks.TASK_STATES.values()
-        task_states_lowercase = list(map(lambda x: x.lower(), task_states))
-        state_shortcuts = Stacks.TASK_STATE_SHORTCUTS
-        if args.tstate and not args.tstate.lower(
-        ) in task_states_lowercase and not args.tstate in state_shortcuts:
-            raise sdb.CommandError(
-                self.name,
-                "'{:s}' is not a valid task state (acceptable states: {:s})".
-                format(args.tstate, ", ".join(task_states)))
-
-        if args.module and Stacks.find_module_memory_segment(
-                args.module)[0] == -1:
-            raise sdb.CommandError(
-                self.name,
-                "module '{:s}' doesn't exist or isn't currently loaded".format(
-                    args.module))
-
-    def _call(self, objs: Iterable[drgn.Object]) -> Iterable[drgn.Object]:
-        # pylint: disable=too-many-locals
-        # pylint: disable=too-many-branches
-        # pylint: disable=too-many-statements
-
+    def validate_context(self) -> None:
         #
-        # As the exception explains the code that follows this statement
-        # only works for linux kernel targets (crash dumps or live systems).
-        # When support for userland is added we can factor the kernel code
-        # that follows into its own function and switch to the correct
+        # This implementation only works for linux kernel targets
+        # (crash dumps or live systems). When support for userland is added we can
+        # refactor the kernel code into its own function and switch to the correct
         # codepath depending on the target.
         #
         if not sdb.get_target_flags() & drgn.ProgramFlags.IS_LINUX_KERNEL:
             raise sdb.CommandError(self.name,
                                    "userland targets are not supported yet")
-        self.validate_args(self.args)
+        self.validate_args()
+
+    def validate_args(self) -> None:
+        if self.args.function:
+            try:
+                func = sdb.get_object(self.args.function)
+                sym = sdb.get_symbol(func.address_of_())
```

Comment:
I saw https://github.com/osandov/drgn/issues/47 - does that mean that `-c func` will only work with non-`static` symbols now?

Reply:
So I was originally trying to look up the symbol with the function name: `prog.symbol(self.args.function)`, which is when I hit the issue with `static` functions. Serapheim came up with this strategy as a workaround. Since switching to using the address, all functions I've tried have worked fine.

Serapheim suggested linking osandov/drgn#47 in a comment in the code to explain why we do this somewhat roundabout lookup, I think I'll do that.

Reply:
tldr: `-c func` should work the same as ever.

Reply:
I think a comment between the `try:` and the `func =` line would be nice explaining that due to https://github.com/osandov/drgn/issues/47 we can't do `sym = sdb.get_symbol(self.args.function)`  so we end up first resolving the address from the object and then getting the symbol from that address (instead of the symbol name).

Reply:
I also added a new test for `-c zthr_procedure` to make sure that static functions continue to work.

Reply:
Good idea!